[BUG] FE restart failed when replaying alter job

<!-- (At least include the following, feel free to add if you have more content) -->

### Steps to reproduce the behavior (Required)
1. create a table
```
CREATE TABLE aggregate_table_with_null (k1 decimal(7,2),k2 int,k3 decimal(30,17),v1 int sum,v2 int min,v3 int max) AGGREGATE KEY (k1, k2, k3) DISTRIBUTED BY HASH(k1, k2, k3) BUCKETS 32 PROPERTIES ( "replication_num" = "3", "in_memory" = "false", "storage_format" = "V2" );
```

2. create a mv on that table
```
create materialized view aggregate_table_with_null_mv as select k1, sum(v1), min(v2), max(v3) from aggregate_table_with_null group by k1;
```

3.  restart FE immediately after step 2.

4. wait for the completion of the alter job.

5. restart FE.

### Expected behavior (Required)

step 5 should succeed.

### Real behavior (Required)
```
 com.starrocks.journal.JournalInconsistentException: failed to load journal type 121
        at com.starrocks.persist.EditLog.loadJournal(EditLog.java:1349)
        at com.starrocks.server.GlobalStateMgr.replayJournalInner(GlobalStateMgr.java:2472)
        at com.starrocks.server.GlobalStateMgr.replayJournal(GlobalStateMgr.java:2421)
        at com.starrocks.leader.Checkpoint.replayAndGenerateGlobalStateMgrImage(Checkpoint.java:208)
        at com.starrocks.leader.Checkpoint.createImage(Checkpoint.java:193)
        at com.starrocks.leader.Checkpoint.runAfterCatalogReady(Checkpoint.java:108)
        at com.starrocks.common.util.FrontendDaemon.runOneCycle(FrontendDaemon.java:72)
        at com.starrocks.common.util.Daemon.run(Daemon.java:107)
Caused by: java.lang.NullPointerException: 12136
        at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:921)
        at com.starrocks.alter.RollupJobV2.onFinished(RollupJobV2.java:699)
        at com.starrocks.alter.RollupJobV2.replayFinished(RollupJobV2.java:858)
        at com.starrocks.alter.RollupJobV2.replay(RollupJobV2.java:893)
        at com.starrocks.alter.AlterHandler.replayAlterJobV2(AlterHandler.java:204)
        at com.starrocks.alter.MaterializedViewHandler.replayAlterJobV2(MaterializedViewHandler.java:912)
        at com.starrocks.persist.EditLog.loadJournal(EditLog.java:857)
```

### StarRocks version (Required)
 - 3.1+

--- a/fe/fe-core/src/main/java/com/starrocks/alter/LakeTableAlterMetaJob.java
+++ b/fe/fe-core/src/main/java/com/starrocks/alter/LakeTableAlterMetaJob.java
@@ -38,6 +38,11 @@ public class LakeTableAlterMetaJob extends LakeTableAlterMetaJobBase {
     @SerializedName(value = "metaValue")
     private boolean metaValue;
 
+    // for deserialization
+    public LakeTableAlterMetaJob() {
+        super(JobType.SCHEMA_CHANGE);
+    }
+
     public LakeTableAlterMetaJob(long jobId, long dbId, long tableId, String tableName,
                                  long timeoutMs, TTabletMetaType metaType, boolean metaValue) {
         super(jobId, JobType.SCHEMA_CHANGE, dbId, tableId, tableName, timeoutMs);

--- a/fe/fe-core/src/main/java/com/starrocks/alter/LakeTableAlterMetaJobBase.java
+++ b/fe/fe-core/src/main/java/com/starrocks/alter/LakeTableAlterMetaJobBase.java
@@ -71,6 +71,10 @@ public abstract class LakeTableAlterMetaJobBase extends AlterJobV2 {
     private Map<Long, Long> commitVersionMap = new HashMap<>();
     private AgentBatchTask batchTask = null;
 
+    public LakeTableAlterMetaJobBase(JobType jobType) {
+        super(jobType);
+    }
+
     public LakeTableAlterMetaJobBase(long jobId, JobType jobType, long dbId, long tableId,
                                      String tableName, long timeoutMs) {
         super(jobId, jobType, dbId, tableId, tableName, timeoutMs);

--- a/fe/fe-core/src/main/java/com/starrocks/alter/LakeTableAsyncFastSchemaChangeJob.java
+++ b/fe/fe-core/src/main/java/com/starrocks/alter/LakeTableAsyncFastSchemaChangeJob.java
@@ -65,6 +65,11 @@ public class LakeTableAsyncFastSchemaChangeJob extends LakeTableAlterMetaJobBase
     private List<IndexSchemaInfo> schemaInfos;
     private Set<String> partitionsWithSchemaFile = new HashSet<>();
 
+    // for deserialization
+    public LakeTableAsyncFastSchemaChangeJob() {
+        super(JobType.SCHEMA_CHANGE);
+    }
+
     LakeTableAsyncFastSchemaChangeJob(long jobId, long dbId, long tableId, String tableName, long timeoutMs) {
         super(jobId, JobType.SCHEMA_CHANGE, dbId, tableId, tableName, timeoutMs);
         schemaInfos = new ArrayList<>();

--- a/fe/fe-core/src/main/java/com/starrocks/alter/LakeTableSchemaChangeJob.java
+++ b/fe/fe-core/src/main/java/com/starrocks/alter/LakeTableSchemaChangeJob.java
@@ -153,6 +153,11 @@ public class LakeTableSchemaChangeJob extends AlterJobV2 {
     private AtomicBoolean waitingCreatingReplica = new AtomicBoolean(false);
     private AtomicBoolean isCancelling = new AtomicBoolean(false);
 
+    // for deserialization
+    public LakeTableSchemaChangeJob() {
+        super(JobType.SCHEMA_CHANGE);
+    }
+
     public LakeTableSchemaChangeJob(long jobId, long dbId, long tableId, String tableName, long timeoutMs) {
         super(jobId, JobType.SCHEMA_CHANGE, dbId, tableId, tableName, timeoutMs);
     }

--- a/fe/fe-core/src/main/java/com/starrocks/alter/OnlineOptimizeJobV2.java
+++ b/fe/fe-core/src/main/java/com/starrocks/alter/OnlineOptimizeJobV2.java
@@ -43,6 +43,7 @@ import com.starrocks.common.util.concurrent.lock.LockType;
 import com.starrocks.common.util.concurrent.lock.Locker;
 import com.starrocks.load.PartitionUtils;
 import com.starrocks.persist.ReplacePartitionOperationLog;
+import com.starrocks.persist.gson.GsonPostProcessable;
 import com.starrocks.persist.gson.GsonUtils;
 import com.starrocks.privilege.PrivilegeBuiltinConstants;
 import com.starrocks.qe.ConnectContext;
@@ -75,14 +76,14 @@ import java.util.concurrent.Executors;
 import java.util.concurrent.Future;
 import java.util.stream.Collectors;
 
-public class OnlineOptimizeJobV2 extends AlterJobV2 {
+public class OnlineOptimizeJobV2 extends AlterJobV2 implements GsonPostProcessable {
     private static final Logger LOG = LogManager.getLogger(OnlineOptimizeJobV2.class);
 
     // The optimize job will wait all transactions before this txn id finished, then send the optimize tasks.
     @SerializedName(value = "watershedTxnId")
     protected long watershedTxnId = -1;
 
-    private final String postfix;
+    private String postfix;
 
     private static final ExecutorService EXECUTOR = Executors.newCachedThreadPool();
 
@@ -114,6 +115,11 @@ public class OnlineOptimizeJobV2 extends AlterJobV2 {
 
     @SerializedName(value = "optimizeOperation")
     private String optimizeOperation = "";
+
+    // for deserialization
+    public OnlineOptimizeJobV2() {
+        super(JobType.OPTIMIZE);
+    }
 
     public OnlineOptimizeJobV2(long jobId, long dbId, long tableId, String tableName, long timeoutMs,
                                OptimizeClause optimizeClause) {
@@ -784,5 +790,10 @@ public class OnlineOptimizeJobV2 extends AlterJobV2 {
                         context.getState().getErrorMessage(), DebugUtil.printId(context.getQueryId()), sql);
             throw new AlterCancelException(context.getState().getErrorMessage());
         }
+    }
+
+    @Override
+    public void gsonPostProcess() throws IOException {
+        this.postfix = "_" + jobId;
     }
 }

--- a/fe/fe-core/src/main/java/com/starrocks/alter/OptimizeJobV2.java
+++ b/fe/fe-core/src/main/java/com/starrocks/alter/OptimizeJobV2.java
@@ -72,7 +72,7 @@ public class OptimizeJobV2 extends AlterJobV2 implements GsonPostProcessable {
     @SerializedName(value = "watershedTxnId")
     protected long watershedTxnId = -1;
 
-    private final String postfix;
+    private String postfix;
 
     @SerializedName(value = "tmpPartitionIds")
     private List<Long> tmpPartitionIds = Lists.newArrayList();
@@ -100,6 +100,11 @@ public class OptimizeJobV2 extends AlterJobV2 implements GsonPostProcessable {
 
     @SerializedName(value = "optimizeOperation")
     private String optimizeOperation = "";
+
+    // for deserialization
+    public OptimizeJobV2() {
+        super(JobType.OPTIMIZE);
+    }
 
     public OptimizeJobV2(long jobId, long dbId, long tableId, String tableName, long timeoutMs,
                          OptimizeClause optimizeClause) {
@@ -764,9 +769,7 @@ public class OptimizeJobV2 extends AlterJobV2 implements GsonPostProcessable {
 
     @Override
     public void gsonPostProcess() throws IOException {
-        if (jobState != JobState.PENDING) {
-            return;
-        }
+        this.postfix = "_" + jobId;
     }
 
     @Override

--- a/fe/fe-core/src/main/java/com/starrocks/alter/RollupJobV2.java
+++ b/fe/fe-core/src/main/java/com/starrocks/alter/RollupJobV2.java
@@ -176,6 +176,11 @@ public class RollupJobV2 extends AlterJobV2 implements GsonPostProcessable {
     private AtomicBoolean waitingCreatingReplica = new AtomicBoolean(false);
     private AtomicBoolean isCancelling = new AtomicBoolean(false);
 
+    // for deserialization
+    public RollupJobV2() {
+        super(JobType.ROLLUP);
+    }
+
     public RollupJobV2(long jobId, long dbId, long tableId, String tableName, long timeoutMs,
                        long baseIndexId, long rollupIndexId, String baseIndexName, String rollupIndexName,
                        int rollupSchemaVersion, List<Column> rollupSchema, Expr whereClause, int baseSchemaHash,

--- a/fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeJobV2.java
+++ b/fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeJobV2.java
@@ -193,6 +193,7 @@ public class SchemaChangeJobV2 extends AlterJobV2 {
         super(jobId, JobType.SCHEMA_CHANGE, dbId, tableId, tableName, timeoutMs);
     }
 
+    // for deserialization
     private SchemaChangeJobV2() {
         super(JobType.SCHEMA_CHANGE);
     }

